src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -236,7 +239,9 @@ private void OnSourceItemChanged(IProjectSubscriptionUpdate e, IProjectItemSchem
 
             foreach (var itemType in e.ProjectChanges.Where(changes => (itemTypesChanged || changes.Value.Difference.AnyChanges) && _itemTypes.Contains(changes.Key)))
             {
-                var items = itemType.Value.After.Items.Select(item => (item.Value[FullPath], GetLink(item.Value), GetCopyType(item.Value)));
+                var items = itemType.Value.After.Items
+                    .Select(item => (GetFullPath(item.Value), GetLink(item.Value), GetCopyType(item.Value)))
+                    .Where(tuple => tuple.Item1 != null);
```

Comment:
❔ Can you briefly explain how this was interfering with the XAML check, and why it's better than the workaround I was using previously?